lib/core/src/Cardano/Wallet/Primitive/Types/UTxOIndex/Internal.hs

```diff
@@ -626,26 +700,31 @@ indexIsComplete i = F.all hasEntry $ Map.toList $ universe i
 --
 indexIsMinimal :: forall u. Ord u => UTxOIndex u -> Bool
 indexIsMinimal i = F.and
```

Comment:
Suggestion (after review with @sevanspowell):

Make the invariant checking stronger by verifying that:
```
indexSingletons ⊆ indexAll
indexPairs      ⊆ indexAll
indexSingletons ∩ indexPairs = ∅
```

Reply:
Fixed in f730c1568e71853787528b02ee2578326fb25258.